### PR TITLE
feat(compose): Introduce create, start, stop subcommands

### PR DESCRIPTION
**Pull request by LucaSeri**
### Prerequisite checklist

  - [x] Read the [contribution guidelines](https://unikraft.org/docs/contributing/) regarding submitting new changes to the project;
  - [x] Tested your changes against relevant architectures and platforms;
  - [x] Ran `make fmt` on your commit series before opening this PR;
  - [ ] Updated relevant documentation.

### Description of changes

Add the ``compose create``, ``start`` and ``stop`` subcommands which provide higher granularity to the state of a project.

Also refactors ``up`` because it is now trivially a ``create`` followed by a ``start``.

This was all mostly achieved by splitting the previous code from ``up`` into multiple parts.

### CODE REVIEW

File: internal/cli/kraft/compose/create/create.go [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+			}
+		}
+		if alreadyRunning {
+			continue
+		}
+
+		driver := "bridge"
+		if network.Driver != "" {
+			driver = network.Driver
+		}
```
Comment (nderjung) [inline internal/cli/kraft/compose/create/create.go]: I wonder if we shouldn't expose this as a `const`/`var` like `DefaultNetworkDriver` and then retrieved from the relevant platform code.

File: internal/cli/kraft/compose/create/create.go [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+
+	machines, err := machineController.List(ctx, &machineapi.MachineList{})
+	if err != nil {
+		return err
+	}
+
+	for _, service := range project.Services {
+		alreadyCreated := false
+		for _, machine := range machines.Items {
+			if service.Name == machine.Name {
```
Comment (nderjung) [inline internal/cli/kraft/compose/create/create.go]: Better to use less indentation where possible, switch this logic to:
```go
if service.Name != machine.Name {
  continue
}
```

File: internal/cli/kraft/compose/create/create.go [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+				if machine.Status.State == machineapi.MachineStateRunning || machine.Status.State == machineapi.MachineStateCreated {
+					alreadyCreated = true
```
Comment (nderjung) [inline internal/cli/kraft/compose/create/create.go]:
```go
if machine.Status.State == machineapi.MachineStateRunning || machine.Status.State == machineapi.MachineStateCreated {
  alreadyCreated = true
  break
}

rmOpts := ...
```

File: internal/cli/kraft/compose/create/create.go [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+		}
+		if service.Image == "" {
+			if err := buildService(ctx, service); err != nil {
+				return err
+			}
+		} else {
+			if err := ensureServiceIsPackaged(ctx, service); err != nil {
+				return err
+			}
+		}
```
Comment (nderjung) [inline internal/cli/kraft/compose/create/create.go]: `else if`

File: internal/cli/kraft/compose/create/create.go [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+			log.G(ctx).WithError(err).Errorf("failed to run service %s", service.Name)
+		}
+
+		if machine, err := machineController.Get(ctx, &machineapi.Machine{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: service.Name,
+			},
+		}); err == nil && machine.Status.State == machineapi.MachineStateCreated {
+			projectMachines = append(projectMachines, machine.ObjectMeta)
+		}
```
Comment (nderjung) [inline internal/cli/kraft/compose/create/create.go]: Potentially uncaught error here

File: internal/cli/kraft/compose/create/create.go [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+	if err != nil {
+		return err
+	}
+
+	log.G(ctx).Infof("creating service %s...", service.Name)
+
+	networks := []string{}
+	for name, network := range service.Networks {
+		networkArg := fmt.Sprintf("%s:%s", project.Networks[name].Name, network.Ipv4Address)
+		networks = append(networks, networkArg)
```
Comment (nderjung) [inline internal/cli/kraft/compose/create/create.go]: Nitpick: this can be inlined.

File: internal/cli/kraft/compose/stop/stop.go [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+	machines, err := machineController.List(ctx, &machineapi.MachineList{})
+	if err != nil {
+		return err
+	}
+
+	machinesToStop := []string{}
+	for _, service := range project.Services {
+		for _, machine := range machines.Items {
+			if service.Name == machine.Name {
+				if machine.Status.State == machineapi.MachineStateRunning {
```
Comment (nderjung) [inline internal/cli/kraft/compose/stop/stop.go]: `if [...] && [...] {`

**Review by nderjung — COMMENTED** [review]
Looks good! Just a few minor comments

File: internal/cli/kraft/compose/create/create.go (hunk 1 above)
Comment (LucaSeri) [reply internal/cli/kraft/compose/create/create.go]: An alternative could be to look into ``hostSupportedStrategies`` and pick the first available?

**Review by nderjung — APPROVED** [review]
Awesome, thanks!

**Pull request merged**